### PR TITLE
Update library loading search paths on z/OS

### LINKED ISSUE #1405: Request for Java 17 documentation update - library loading search paths on z/OS
Will Smith informed us about an intentional change in behaviour from Java 11 to Java 17 and wanted to know the topic that can be updated with this information. This documentation update is important from the Java on z/OS support perspective because multiple customers have reported problems when they moved to Java 17 and their application stopped working.
 
On z/OS, when the JVM loads an application native library (DLL), it searches for the library in the path from java system property java.library.path. From jdk17+, the search is based only off the java.library.path system property value, obtained when the JVM starts. Changing java.library.path after JVM startup doesn't change the search. The default java.library.path is derived from environment variable LIBPATH. If the java.library.path is set on the command line, this value (only) is used for the search and LIBPATH is not searched for application native libraries.
 
Before this change, for Java 11 and earlier, the JVM searched for application native libraries using both java.library.path and LIBPATH.

After analysing the OpenJ9 document topics, we came to the conclusion that we can mention this change in the migrating topic there - [Migrating from Java 11 to Java 17 - (eclipse.dev)](https://eclipse.dev/openj9/docs/migrating11to17/).

[Java 17 doc update-zOS.pdf](https://github.com/user-attachments/files/16929365/Java.17.doc.update-zOS.pdf)

### PR DESCRIPTION
**Pull request by Sreekala-Gopakumar**
https://github.com/eclipse-openj9/openj9-docs/issues/1405

Updated the Migrating from Java 11 to Java 17 topic with the change in the search for the library behavior.

Closes #1405
Signed-off-by: Sreekala Gopakumar sreekala.gopakumar@ibm.com

### CODE REVIEW

**Comment by Sreekala-Gopakumar** [comment]
Jenkins doc stage

**Comment by Sreekala-Gopakumar** [comment]
Jenkins doc stage

File: docs/migrating11to17.md [hunk 1]
```diff
@@ -45,5 +45,7 @@ The following table lists the new OpenJ9 features and notable changes with the O
...
+|On z/OS, when the VM loads an application native library (DLL), it searches for the library in the path that is specified in the `-Djava.library.path` system property or the `LIBPATH` environment variable. In Java 11 and earlier versions, the VM searched for the application native libraries by using both the `-Djava.library.path` system property and the `LIBPATH` environment variable. From Java 17 onwards, the search is based on only the `-Djava.library.path` system property value that is obtained when the VM starts. Changing the `-Djava.library.path` system property value after VM startup doesn't change the search. If the path is set on the command line by using the `-Djava.library.path` system property, only this path is used for the search. The path specified by the `LIBPATH` variable is not used to search for the application native libraries. The default value of the  `-Djava.library.path` system property is still derived from the `LIBPATH` environment variable.  |  0.47.0  |
```
Comment (pshipton) [inline docs/migrating11to17.md:49]: The change to only look at the `java.library.path` is not specific to z/OS, it applies to all platforms. `LIBPATH` is specific to z/OS and AIX, Linux uses LD_LIBRARY_PATH, Windows uses PATH.

**Comment by Sreekala-Gopakumar** [comment]
Jenkins doc stage

**Comment by Sreekala-Gopakumar** [comment]
will.smith@uk.ibm.com @pshipton - As discussed, I have updated the Migrating from Java 11 to Java 17 topic. Please check and let me know if any changes are required. Thanks!

https://eclipse-openj9.github.io/openj9-docs-staging/1410/migrating11to17/

**Comment by pshipton** [comment]
Not sure if my comment was seen.

The change to only look at the java.library.path is not specific to z/OS, it applies to all platforms. LIBPATH is specific to z/OS and AIX, Linux uses LD_LIBRARY_PATH, Windows uses PATH.

Also the change is not in the 0.47 release, it's in jdk17 from the first release. You can lookup the version where jdk17 was added, or leave out the OpenJ9 release version since it's not relevant.

**Comment by Sreekala-Gopakumar** [comment]
Jenkins doc stage

File: docs/migrating11to17.md [hunk 2]
```diff
@@ -45,5 +45,7 @@ The following table lists the new OpenJ9 features and notable changes with the O
...
+|When the VM loads an application native library (DLL), it searches for the library in the path that is specified in the `-Djava.library.path` system property or in the following environment variables:<ul><li>`LIBPATH` - z/OS&reg; and AIX&reg;</li><li>`LD_LIBRARY_PATH` - Linux&reg;<li>`PATH` - Windows&trade;</li></ul><br>In Java 11 and earlier versions, the VM searched for the application native libraries by using both the `-Djava.library.path` system property and the environment variable.<br>From Java 17 onwards, the search is based on only the `-Djava.library.path` system property value that is obtained when the VM starts. Changing the `-Djava.library.path` system property value after VM startup doesn't change the search. If the path is set on the command line by using the `-Djava.library.path` system property, only this path is used for the search. The path that is specified by the variable is not used to search for the application native libraries.<br>The default value of the `-Djava.library.path` system property is still derived from the environment variable.  |  0.29.1   |
```
Comment (pshipton) [inline docs/migrating11to17.md:49]: `-Djava.library.path` is a command line option. For all usage except for "If the path is set on the command line by using the `-Djava.library.path` system property", it should just be `java.library.path`. 
i.e. "it searches for the library in the path that is specified in the `java.library.path` system property"

Comment (pshipton) [reply docs/migrating11to17.md:49]: "by using the -Djava.library.path system property"
-> "by using the -Djava.library.path option"

Comment (pshipton) [reply docs/migrating11to17.md:49]: I assume "the variable" in "The path that is specified by the variable is not used to search for the application native libraries." refers to the LIBPATH, etc. variables? This isn't clear.

Comment (Sreekala-Gopakumar) [reply docs/migrating11to17.md:49]: If I say "... specified by the environment variables...", will that be clear?

**Comment by Sreekala-Gopakumar** [comment]
Jenkins doc stage

File: docs/migrating11to17.md (hunk 2 above)
Comment (pshipton) [reply docs/migrating11to17.md]: Sounds good.

Comment (pshipton) [reply docs/migrating11to17.md]: It should be "environment variable" rather than "environment variables" (no 's').

**Comment by smithwil** [comment]
[Thanks @pshipton for notifying me about this.] @Sreekala-Gopakumar  I have reviewed the new user guide text and I think it is good, I don't have any further changes. Thanks for the edit.

**Comment by Sreekala-Gopakumar** [comment]
Jenkins doc stage

**Comment by Sreekala-Gopakumar** [comment]
@pshipton - Please check and confirm. Thanks!

**Review by pshipton — APPROVED** [review]

**Pull request merged**